packages/subgraph/tests/helpers/method-classes.ts

```diff
@@ -0,0 +1,291 @@
+  VotingSettingsUpdated
+} from '../../generated/templates/TokenVoting/TokenVoting';
+import {
+  VOTER_OPTIONS,
+  VOTE_OPTIONS,
+  VOTING_MODES,
+  VOTING_MODE_INDEXES
+} from '../../src/utils/constants';
+import {
+  ADDRESS_ONE,
+  ALLOW_FAILURE_MAP,
+  CONTRACT_ADDRESS,
+  CREATED_AT,
+  DAO_ADDRESS,
+  END_DATE,
+  MIN_VOTING_POWER,
+  PROPOSAL_ENTITY_ID,
+  PROPOSAL_ID,
+  SNAPSHOT_BLOCK,
+  START_DATE,
+  SUPPORT_THRESHOLD,
+  TOTAL_VOTING_POWER,
+  TWO,
+  VOTING_MODE,
+  ZERO,
+  MIN_PARTICIPATION,
+  MIN_DURATION,
+  ONE,
+  DAO_TOKEN_ADDRESS,
+  STRING_DATA
+} from '../constants';
+import {
+  createNewProposalCreatedEvent,
+  createNewProposalExecutedEvent,
+  createNewVoteCastEvent,
+  createNewVotingSettingsUpdatedEvent,
+  getProposalCountCall
+} from '../token/utils';
+import {createGetProposalCall, createTotalVotingPowerCall} from '../utils';
+
+class ERC20ContractMethods extends ERC20Contract {
+  withDefaultValues(): ERC20ContractMethods {
+    this.id = Address.fromHexString(CONTRACT_ADDRESS).toHexString();
+    this.name = 'Test Token';
+    this.symbol = 'TT';
+    this.decimals = 18;
+
+    return this;
+  }
+}
+class TokenVotingVoterMethods extends TokenVotingVoter {
+  withDefaultValues(): TokenVotingVoterMethods {
+    this.id = Address.fromHexString(CONTRACT_ADDRESS)
+      .toHexString()
+      .concat('_')
+      .concat(ADDRESS_ONE);
+    this.address = ADDRESS_ONE;
+    this.plugin = Address.fromHexString(CONTRACT_ADDRESS).toHexString();
+    this.lastUpdated = BigInt.fromString(ZERO);
+
+    return this;
+  }
+}
+
+class TokenVotingProposalMethods extends TokenVotingProposal {
+  withDefaultValues(): TokenVotingProposalMethods {
+    this.id = PROPOSAL_ENTITY_ID;
+
+    this.dao = DAO_ADDRESS;
+    this.plugin = Address.fromHexString(CONTRACT_ADDRESS).toHexString();
+    this.proposalId = BigInt.fromString(PROPOSAL_ID);
+    this.creator = Address.fromHexString(ADDRESS_ONE);
+
+    this.open = true;
+    this.executed = false;
+
+    this.votingMode = VOTING_MODE;
+    this.supportThreshold = BigInt.fromString(SUPPORT_THRESHOLD);
+    this.minVotingPower = BigInt.fromString(MIN_VOTING_POWER);
+    this.startDate = BigInt.fromString(START_DATE);
+    this.endDate = BigInt.fromString(END_DATE);
+    this.snapshotBlock = BigInt.fromString(SNAPSHOT_BLOCK);
+
+    this.yes = BigInt.fromString(ZERO);
+    this.no = BigInt.fromString(ZERO);
+    this.abstain = BigInt.fromString(ZERO);
+    this.castedVotingPower = BigInt.fromString(ZERO);
+
+    this.totalVotingPower = BigInt.fromString(TOTAL_VOTING_POWER);
+    this.allowFailureMap = BigInt.fromString(ALLOW_FAILURE_MAP);
+    this.createdAt = BigInt.fromString(CREATED_AT);
+    this.creationBlockNumber = BigInt.fromString(ZERO);
+    this.potentiallyExecutable = false;
+
+    return this;
+  }
+
+  // calls
+  // (only read call from contracts related to this)
+  mockCall_getProposal(actions: ethereum.Tuple[]): void {
+    if (!this.yes || !this.no || !this.abstain) {
```

Comment:
Couldn't a proposal have `0` yes votes (correct)... and fail because of the `!this.yes` condition?

Reply:
it is not about 0, it is about it being null or not